wincent/liferay-portal
# LPS-112298 Update Liferay.Util.selectEntity utility

Hey @wincent ,

Going into this task, I was planning to make new versions of `selectEntity` and `selectEntityHandler` utilities. However, I found that we can massively simplify code, by only using `openModal`, exending its API with the `onSelect` callback. In this PR, we are updating usages in `fragment-web`.

To test fragment entry move 1
1. Open Control Panel → Site → Site Builder → Page Fragments
1. Add two sample collections and a sample fragment component in the first
1. Click component dropdown → Move
1. Select second collection

To test fragment entry move 2
- The same as previous, except use management toolbar instead of component dropdown (select a card to open toolbar)

To test fragment entry copy 1
1. On Page Fragments page, open Basic Components
1. Click component dropdown → Copy To
1. Select previously created sample collection

To test fragment entry copy 2
- The same as previous, except use management toolbar

I was not able to test fragment composition move, I could not find this in UI. WIP maybe? You need to create fragment composition somehow, and then it is the same dropdown as in the first test. I am pretty confident this would work, it is the same change as in other usages. Product team should be able to test this.

Sidenote: What bugged me a bit is that `onSelect` seems deceptively generic, similar to unambiguous `onClose`. However, it requires very specific structure to work, `.selector-button` elements in content with data attributes. It inherits this from `selectEntity` util, which has equally hardcoded implementation requirements, that user of utility must know. Maybe it would make sense to have a more generic API, at least with custom trigger class and custom attribute. But, I opted to to keep it closer to original way of using it, at least for now. We can always easily extend API later on, if we want to.

Please review the code.

Thank you!

/cc @jbalsas
